### ISSUE TITLE
False negative when a class-scoped covariant or contravariant TypeVar used in invariant base class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3981]
The following code demonstrates a hole in type checking:

```python
from typing import TypeVar

T = TypeVar("T", covariant=True)

class MyList(list[T]): pass

def append_float(y: MyList[float]):
    y.append(1.0)

x: MyList[int] = MyList([1, 2, 3])
append_float(x)
```

The problem here is that `T` is covariant, but it's being used in a base class `list` whose corresponding type parameter is invariant. This means usage of the resulting subclass `MyList` can now violate the variance of its base class.

A type checker should catch this case and report an error.

This was discovered in this [typeshed issue](https://github.com/python/typeshed/pull/8781).

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
@hauntsaninja noted that my check did not take into account mismatch between covariant and contravariant and vice versa. I've updated the check accordingly.

**Comment by erictraut** [comment]
This issue is addressed in pyright 1.1.273, which I just published. It will also be included in a future release of pylance.
